pkg/account/domain/account.go

```diff
@@ -236,3 +304,34 @@ func (a *AccountV2) resetDefaultFilter(targetFilter proto.FilterTargetType, envi
 		}
 	}
 }
+
+func validate(a *AccountV2) error {
```

Comment:
It's better to validate all params before change, we do all validation in [pkg/account/api/validation.go](https://github.com/bucketeer-io/bucketeer/pull/1321/files#diff-0de67f64830366bcb37ac25ac98abe4f2e03ef06f8ec8a05f78798e5b18b3957)

Reply:
You already validate params [here](https://github.com/bucketeer-io/bucketeer/pull/1321/files#diff-0de67f64830366bcb37ac25ac98abe4f2e03ef06f8ec8a05f78798e5b18b3957R461), so you don't need to validate again。

Reply:
hmm, currently in the domain level of feature, the validation is also called after the update, https://github.com/bucketeer-io/bucketeer/blob/main/pkg/feature/domain/feature.go#L1039

Can you tell me why? @cre8ivejp @kentakozuka

Reply:
Generally, validations in API layer check if the request is simply valid(e.g. empty value or invalid format), while validations in domain layer validate more like business logic (e.g. check if the flag dependencies are valid regarding to other existing data in DB). That's why we have validations in domain later too.
(But it might also true that the definition above is not strictly respected.)

Reply:
Same opinion.
It's ok to do a simple empty validation in the API layer. More complex validations should be done in the domain layer.

Reply:
I see, for now I want extra validation after update in domain layer so if this function is being reused, the business logic is kept inside domain layer. In the future when we removing mysql depend in API layer, maybe we only do validation in domain layer. Wdyt? @Ubisoft-potato

Reply:
> I see, for now I want extra validation after update in domain layer so if this function is being reused, the business logic is kept inside domain layer. In the future when we removing mysql depend in API layer, maybe we only do validation in domain layer. Wdyt? @Ubisoft-potato

Good thought! 
Let's keep this domain layer validation.